### ISSUE TITLE
Add a results count and pagnination to top of dataset cards in catalog search

### ISSUE THREAD

**Issue by j9recurses** [DataSF/open-data-explorer#324]
In the dataset catalog page,
- add a top pagination bar
- Add a total number of results to the top 
<img width="1280" alt="screen shot 2017-06-21 at 4 44 57 pm" src="https://user-images.githubusercontent.com/6549842/27411438-9f5bb798-56a2-11e7-960f-82ae2c82295c.png">

**Comment by jasonlally** [comment]
Total number of results is in there, but it's not obvious. In the previous version before react-instantsearch we had a more prominent count, but it's just a matter of some styling. We can also drop the speed of the search. That's more a marketing thing for Algolia than useful information to the end user.

<img width="271" alt="screen shot 2017-06-22 at 8 45 28 am" src="https://user-images.githubusercontent.com/534176/27443600-7110a214-5728-11e7-805a-6e58434a6646.png">

**Comment by jasonlally** [comment]
Added clearer number of results. Avoiding adding pagination at top for now. I'd like to observe use a bit more on this as the best practices suggest having a single pagination at the bottom. 

It also clutters the top a bit and with quick search, pagination may not be used all that often. We can capture this in google analytics.

I've updated it so clicking next will refocus at the top of the results, which is what most users expect from say Amazon or Google. I think that was what was most confusing in the previous iteration.
